test/transcoding/SPV_INTEL_fpga_dsp_control/negative.ll

```diff
@@ -0,0 +1,85 @@
+; template<typename Function>
+; [[intel::prefer_dsp]]
+; [[intel::propagate_dsp_preference]]
+; void math_prefer_dsp_propagate(Function f)
+; {
+;   f();
+; }
+
+; int main() {
+;   math_prefer_dsp_propagate([]() {
+;     int a = 0;
+;     a += 1;
+;   });
+
+; return 0;
+; }
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv %t.spv --to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV-NOT: Capability MathOpDSPModeINTEL
```

Comment:
Negative test could be added inside the existing tests, just by checking with another check prefix. I think it is simpler and doesn't require copying the same IR. Feel free to ignore this suggestion.